api/src/main/java/org/openmrs/module/cmrreports/api/reporting/SetupPECGReport.java

```diff
@@ -0,0 +1,118 @@
+		dsd.addColumn("2All5-9", "PECG: Currently on ART: Patients between 5 and 9 years", new Mapped<CohortIndicator>(Indicators.getPatientBetween5And9YearsIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		dsd.addColumn("2All10-14", "PECG: Currently on ART: Patients between 10 and 14 years", new Mapped<CohortIndicator>(Indicators.getPatientBetween10And14YearsIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		dsd.addColumn("2All15-19", "PECG: Currently on ART: Patients between 15 and 19 years", new Mapped<CohortIndicator>(Indicators.getPatientBetween15And19YearsIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		dsd.addColumn("2All20-24", "PECG: Currently on ART: Patients between 20 and 24 years", new Mapped<CohortIndicator>(Indicators.getPatientBetween20And24YearsIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		dsd.addColumn("2All25-49", "PECG: Currently on ART: Patients between 25 and 49 years", new Mapped<CohortIndicator>(Indicators.getPatientBetween25And49YearsIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		dsd.addColumn("2All50-Above", "PECG: Currently on ART: Patients between 50 years and above", new Mapped<CohortIndicator>(Indicators.getPatientBetween50AndAboveIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		
+		// Male
+		int i = 0;
+		for (AgeCohortDefinition ageCohort : Cohorts.getAllAgeRanges()) {
+			dsd.addColumn("2M" + i, "Males:PECG: Currently on ART by age and sex: " + ageCohort.getName(), new Mapped<CohortIndicator>(Indicators.getPatientInYearRangeIndicator(ageCohort, Cohorts.getMales()), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+			i++;
+		}
+		
+		// Females
+		int j = 0;
+		for (AgeCohortDefinition ageCohort : Cohorts.getAllAgeRanges()) {
+			dsd.addColumn("2F" + j, "Females:PECG: Currently on ART by age and sex: " + ageCohort.getName(), new Mapped<CohortIndicator>(Indicators.getPatientInYearRangeIndicator(ageCohort, Cohorts.getFemales()), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+			j++;
+		}
+		dsd.addColumn("2All", "PECG: Currently on ART", new Mapped<CohortIndicator>(Indicators.getAllPatientsIndicator(), ParameterizableUtil.createParameterMappings("startDate=${startDate},endDate=${endDate}")), "");
+		
+	}
+	
+	private void setUpProperties() {
+		HIVProgram = Context.getProgramWorkflowService().getProgram(1);
+		HIVPrograms.add(HIVProgram);
+		UPECVisitType = Context.getVisitService().getVisitTypeByUuid("a7c2aaf0-c4e5-4310-aa94-07c7fe6a331a");
```

Comment:
@rubailly Visit type UPECVisitType is not part of our current project. We should add it in the cameroon-bahmni-metadata repository.

Reply:
We should call it VISIT_TYPE_ART_CONSULTATION and add the translation in the locale file.